### ISSUE TITLE
multi_scan test retrying does not work with multiple devices

### ISSUE THREAD

**Issue by akoller-vs** [lyndsey-ferguson/fastlane-plugin-test_center#263]
### New Issue Checklist

- [x] Updated `fastlane-plugin-test_center` to the latest version
- [x] I read the [README.md](https://github.com/lyndsey-ferguson/fastlane-plugin-test_center/blob/master/README.md)
- [x] I reviewed the [example(s)](https://github.com/lyndsey-ferguson/fastlane-plugin-test_center/blob/master/README.md) for the action(s) I am using
- [x] I have removed any sensitive data such as passwords, authentication tokens, or anything else I do not want to world to see

### Issue Description

Running `multi_scan` with `try_count` 1+ and with multiple devices does not properly run failed tests. For example, if I run on iPhone 8 (13.5) and have 22 tests, 1 failure, and then run the same tests on iPhone 8 (12.4) and have 22 tests, 0 failures, I would expect to see the 1 failed test on iPhone 8 (13.5) to be reran. Instead, `multi_scan` recognizes there are failed tests and starts a second scan, but no tests are included in the try.

I have noticed that the failed tests will be properly rerun if they fail on all selected devices (both iPhone 8 (13.5) and iPhone 8 (12.4) in my scenario). I also thought I've previous see this work the opposite way, in that if a test failed on 1 device, it would be reran on all devices. That is fine, although not optimal, but skipping failed tests when they only fail on a subset of devices defeats the purpose of testing on multiple devices.

```
multi_scan(
    scheme: scheme, 
    devices: ["iPhone 8 (13.3)", "iPhone 8 (12.4)"],
    code_coverage: false, 
    derived_data_path: "./DerivedData", 
    output_directory: "./reports", 
    formatter: "xcpretty-json-formatter", 
    try_count: 3,
    testrun_completed_block: test_run_block
  )
```

<details>
  <pre>

SimDevice: iPhone 8 (4D398D91-CEB3-47C8-8BF8-2BEE152D9E9E, iOS 13.5
...
Executed 22 tests, with 1 failures (1 unexpected) in 451.237 (451.968) seconds  
...
SimDevice: iPhone 8 (D4746D98-587F-4C12-B5EF-ABDF6C88F346, iOS 12.4
...
 Executed 22 tests, with 0 failures (0 unexpected) in 365.800 (365.826) seconds
...
** TEST EXECUTE FAILED **                                                                                                                                       │
[16:16:35]: Exit status: 65                                                                                                                                     │
+--------------------+----+                                                                                                                                     │
|      Test Results       |                                                                                                                                     │
+--------------------+----+                                                                                                                                     │
| Number of tests    | 44 |                                                                                                                                     │
| Number of failures | 1 |                                                                                                                                     │
+--------------------+----+  

SimDevice: iPhone 8 (4D398D91-CEB3-47C8-8BF8-2BEE152D9E9E, iOS 13.5
...
Executed 0 tests, with 0 failures (0 unexpected) in 0.000 (0.001) seconds
...
SimDevice: iPhone 8 (D4746D98-587F-4C12-B5EF-ABDF6C88F346, iOS 12.4
...
 Executed 0 tests, with 0 failures (0 unexpected) in 0.000 (0.000) seconds   
...
+--------------------+---+                                                                                                                                      │
|      Test Results      |                                                                                                                                      │
+--------------------+---+                                                                                                                                      │
| Number of tests    | 0 |                                                                                                                                      │
| Number of failures | 0 |                                                                                                                                      │
+--------------------+---+      

</pre>
</details>

### Environment

<details>
  <pre>

| Key                         | Value                                       |
| --------------------------- | ------------------------------------------- |
| OS                          | 10.15.5                                     |
| Ruby                        | 2.4.2                                       |
| Bundler?                    | true                                        |
| Git                         | git version 2.24.3 (Apple Git-128)          |
| Installation Source         | ~/.rvm/gems/ruby-2.4.2/bin/fastlane         |
| Host                        | Mac OS X 10.15.5 (19F101)                   |
| Ruby Lib Dir                | ~/.rvm/rubies/ruby-2.4.2/lib                |
| OpenSSL Version             | OpenSSL 1.1.1d  10 Sep 2019                 |
| Is contained                | false                                       |
| Is homebrew                 | false                                       |
| Is installed via Fabric.app | false                                       |
| Xcode Path                  | /Applications/Xcode.app/Contents/Developer/ |
| Xcode Version               | 11.5                                        |

| Gem      | Version | Update-Status      |
| -------- | ------- | ------------------ |
| fastlane | 2.152.0 | 🚫 Update available |

| Plugin                                    | Version | Update-Status      |
| ----------------------------------------- | ------- | ------------------ |
| fastlane-plugin-test_center               | 3.11.5  | 🚫 Update available |
| fastlane-plugin-firebase_app_distribution | 0.1.4   | ✅ Up-To-Date       |</pre>
</details>

**Comment by lyndsey-ferguson** [comment]
When running `xcodebuild` with multiple devices, as is the case here, the junit report records the testcase results in the same junit report file. It does this without indicating for which device the tests failed or passed.

`multi_scan` is using the junit report to determine which tests failed and which tests passed. 

In order to fix this, `multi_scan` would have to call `scan` separately for each device and use the generated junit report to control which tests are retried for that device.

**Comment by lyndsey-ferguson** [comment]
@akoller-vs I believe that I have addressed the issue. Can you modify your `Pluginfile` per my instructions below, run `bundle install`, and then run your fastlane again (with the `--verbose` flag)?

Pluginfile:
```
gem 'fastlane-plugin-test_center', :git => "https://github.com/lyndsey-ferguson/fastlane-plugin-test_center.git", :branch => "issue-263-multi-device-parallel-test-run"
```

If there are still problems, please let me know and attach the console output as a text file to this issue (makes it easier for me to review).  If it works, please let me know.

**Comment by lyndsey-ferguson** [comment]
Fixed in [v3.14.5](https://github.com/lyndsey-ferguson/fastlane-plugin-test_center/releases/tag/v3.14.5)
